src/Adapter/Order/CommandHandler/UpdateProductInOrderHandler.php

```diff
@@ -58,54 +61,65 @@ final class UpdateProductInOrderHandler extends AbstractOrderHandler implements
      */
     private $orderProductQuantityUpdater;
 
+    /**
+     * @var ContextStateManager
+     */
+    private $contextStateManager;
+
     /**
      * UpdateProductInOrderHandler constructor.
      *
      * @param OrderProductQuantityUpdater $orderProductQuantityUpdater
+     * @param ContextStateManager|null $contextStateManager
      */
     public function __construct(
-        OrderProductQuantityUpdater $orderProductQuantityUpdater
+        OrderProductQuantityUpdater $orderProductQuantityUpdater,
+        ContextStateManager $contextStateManager = null
     ) {
         $this->orderProductQuantityUpdater = $orderProductQuantityUpdater;
+        $this->contextStateManager = $contextStateManager ?: new ContextStateManager(Context::getContext());
```

Comment:
Not a fan of manually creating this, I understand that it's to avoid a BC break but it's an an adapter class for a handler which is not at all dedicated to be extended It's even `final` so in this case (imho) the BC break is not only acceptable but recommended
@matks what do you think?

Reply:
But it's a recommendation from @matks :sob:

Reply:
I actually suggested this solution 😅  I thought it was an easy and inoffensive way to avoid the BC

Reply:
@Progi1984 I know I read the history That's why I pinged him ^^ I agree that the optional parameter + manual building is sometimes useful especially for Services that are used in legacy classes, or that can be extended or used by modules

But this one just can't because of the `final` so complexifying the constructor for a potential extension use case that can't happen seems overkill and counter productive

Note: the `final` here is a good argument, but I think even some handlers that don't have it - because we forgot it - should be modifiable in the same way or else we can never fix them properly because we are afraid of BC because of a dependency that is always injected via the service DI anyway. Besides in this case manual building is easy but it's not always the case

Finally, and maybe most important, the ContextStateManager has a principle of stacking the overrides when consecutive calls are made one after another, but if another instance is used in different parts of the code they don't share this stack thus the whole stack behaviour can't work and it can cause unexpected behaviours..